pkg/build/builder/docker.go

```diff
@@ -64,16 +60,10 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	glog.V(4).Infof("Starting Docker build from build config %s ...", d.build.Name)
-	var push bool
-
 	// if there is no output target, set one up so the docker build logic
-	// will still work, but we won't push it at the end.
+	// (which requires a tag) will still work, but we won't push it at the end.
 	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name) == 0 {
-		d.build.Spec.Output.To = &kapi.ObjectReference{
-			Kind: "DockerImage",
-			Name: noOutputDefaultTag,
-		}
-		push = false
+		d.build.Status.OutputDockerImageReference = d.build.Name
```

Comment:
> i can't remember the details.

I think [this was the discussion](https://github.com/openshift/origin/pull/5071#discussion_r43018613), found it thanks to some old emails.

Still, I think there's no point in changing that build.Status at this point... what is the point?
The `build.Status.OutputDockerImageReference` should have been correctly set by the controller (by deriving it from `build.Spec.Output.To.Name`).

And we are breaking a "promise" wrt what we pass in the BUILD env var with the marshaled build object.
Before we were setting some fake value in `d.build.Spec.Output.To` so that the builder code, e.g. assemble script) could assume this value is always defined.

Again, there seems to be some logic problem here, as we're really doing:

```go
if A != "" {
   B = X
}
```

While before we had:


```go
if A != "" {
   A = X
}
```